File: src/pages/LoginPage.js
const BasePage = require('./BasePage');

class LoginPage extends BasePage {
  constructor(driver) {
    super(driver);
  }

  // Locators
  get emailField() {
    return '#login-email';
  }

  get passwordField() {
    return '#login-password';
  }

  get loginButton() {
    return '#login-submit-button';
  }

  get errorLabel() {
    return '#auth-error-message';
  }

  get rememberCheckbox() {
    return '#remember-me-checkbox';
  }

  get goToRegisterLink() {
    return '#go-to-register';
  }

  /**
   * Complete login action.
   */
  async login(email, password, remember = false) {
    if (email !== null) {
      await this.clearAndSetValue(this.emailField, email, `Enter login email: ${email}`);
    }
    if (password !== null) {
      await this.clearAndSetValue(this.passwordField, password, 'Enter login password');
    }
    if (remember) {
      await this.click(this.rememberCheckbox, 'Check Remember Me option');
    }
    await this.click(this.loginButton, 'Click Sign In Button');
  }

  /**
   * Gets auth error validation messages.
   */
  async getErrorMessage() {
    return await this.getText(this.errorLabel, 'Get login error validation text');
  }

  /**
   * Navigate to Registration view.
   */
  async navigateToRegister() {
    await this.click(this.goToRegisterLink, 'Click Register Here link');
  }
}

module.exports = LoginPage;

File: src/main/java/com/smartbudget/pages/LoginPage.java
package com.smartbudget.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class LoginPage extends BasePage {
    private By emailField = By.id("login-email");
    private By passwordField = By.id("login-password");
    private By rememberCheckbox = By.id("remember-me-checkbox");
    private By loginSubmitButton = By.xpath("//form[@id='login-form']//button[@type='submit']");
    private By errorMessage = By.id("auth-error-message");
    private By registerLink = By.id("go-to-register");

    public LoginPage(WebDriver driver) {
        super(driver);
    }

    public void login(String email, String password, boolean remember) {
        sendKeys(emailField, email);
        sendKeys(passwordField, password);
        if (remember && driver != null) {
            click(rememberCheckbox);
        }
        click(loginSubmitButton);
    }

    public String getErrorMessage() {
        return getText(errorMessage);
    }

    public void clickRegister() {
        click(registerLink);
    }
}
